### PR TITLE
net: adhere to new naming

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
TODO:
- [x] cleanup queue doc comments
- [x] fixup README of network queue
- [x] mux -> virt
- [x] before merging, check that the binaries are identical before and after the patch

### CODE REVIEW

File: network/queue/queue.c
Comment (wom-bat) [inline network/queue/queue.c:1]: Get rid of this file; include `net_queue_init` as a static inline in queue.h

**Review by wom-bat — CHANGES_REQUESTED** [review]

File: network/queue/queue.c
Comment (Ivan-Velickovic) [reply network/queue/queue.c:1]: done

File: network/components/virt_tx.c [hunk 1]
```diff
@@ -27,8 +27,8 @@ uintptr_t buffer_data_region_cli0_paddr;
 uintptr_t buffer_data_region_cli1_paddr;
 
 typedef struct state {
-    ring_handle_t tx_ring_drv;
-    ring_handle_t tx_ring_clients[NUM_CLIENTS];
+    net_queue_handle_t tx_ring_drv;
```
Comment (wom-bat) [inline network/components/virt_tx.c:30]: `tx_queue_drv` and `tx_quque_clients`

**Review by wom-bat — CHANGES_REQUESTED** [review]
Found some more ring->queue changes needed

**Comment by Ivan-Velickovic** [comment]
Okay, I think I've got all of them, now...

File: include/sddf/network/queue.h [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+#include <stddef.h>
+#include <stdint.h>
+#include <sddf/network/queue.h>
+#include <sddf/network/constants.h>
+#include <sddf/util/fence.h>
+#include <sddf/util/util.h>
+
+typedef struct net_buff_desc {
+    /* offset of buffer within buffer memory region or physical address of buffer */
+    uintptr_t phys_or_offset;
```
[the remark this answers is not in the document]
Comment (wom-bat) [reply include/sddf/network/queue.h:19]: In fact the way these are used assumes a single mapping --- the virtualisers will not work if the region is not contiguous.

**Review by wom-bat — CHANGES_REQUESTED** [review]

**Review by wom-bat — APPROVED** [review]
Looks good now.

File: drivers/network/meson/ethernet.c [hunk 3]
```diff
@@ -42,15 +42,15 @@ _Static_assert((RX_COUNT + TX_COUNT) * sizeof(struct descriptor) <= HW_REGION_SI
...
 
 hw_ring_t rx;
 hw_ring_t tx;
 
-ring_handle_t rx_ring;
-ring_handle_t tx_ring;
+net_queue_handle_t rx_queue;
+net_queue_handle_t tx_queue;
 
 volatile struct eth_mac_regs *eth_mac = (void *)(uintptr_t)0x2000000;
```
[the remark this answers is not in the document]
Comment (Ivan-Velickovic) [reply drivers/network/meson/ethernet.c:55]: It's a virtual address, I don't know why the code is like this is can just use `setvar_vaddr` like everything else I think.

[the remark this answers is not in the document]
Comment (Courtney3141) [reply drivers/network/meson/ethernet.c:55]: I noticed this recently and was planning on updating it.

**Comment by Courtney3141** [comment]
I have just checked the memory barriers in queue.h, everything looks good to me. I built for both the odroidc4 and the imx8mm and ran some benchmarks - everything performed as per usual. I am happy to merge this pull request.

**Pull request merged**